silhouette/app/com/mohiva/play/silhouette/impl/authenticators/CookieAuthenticator.scala

```diff
@@ -282,9 +282,16 @@ class CookieAuthenticatorService(
    * @return The manipulated request header.
    */
   override def embed(cookie: Cookie, request: RequestHeader): RequestHeader = {
+    def mkCookies(cookies: Seq[Cookie]): Cookies = new Cookies {
+      lazy val cookiesByName = cookies.groupBy(_.name).transform((_, cs) => cs.head)
+      override def get(name: String) = cookiesByName.get(name)
+      override def foreach[U](f: Cookie => U): Unit = cookies.foreach(f)
+      override def iterator: Iterator[Cookie] = cookies.iterator
+    }
+
     val filteredCookies = request.cookies.filter(_.name != cookie.name).toSeq
     val combinedCookies = filteredCookies :+ cookie
-    val cookies = Cookies(combinedCookies)
+    val cookies = mkCookies(combinedCookies)
```

Comment:
As mentioned [here](https://github.com/playframework/playframework/pull/9215), this can be replaced by 

```suggestion
    val cookiesByName = combinedCookies.groupBy(_.name).transform((_, cs) => cs.head)
    val cookies = cookieHeaderEncoding.fromMap(combinedCookies)
```

The drawback is that `cookiesByName` can't be lazy in that case.

Reply:
I thought `fromMap` was private

Reply:
you are write, `fromMap` is private. I missed it.